Python/Product/TestAdapter.Executor/TestExecutor.cs

```diff
@@ -532,7 +532,11 @@ sealed class TestRunner : IDisposable {
                         DebugInfo("set " + pythonPath.Key + "=" + pythonPath.Value);
                         DebugInfo(proc.Arguments);
 
-                        _connected.WaitOne();
+                        // If there's an error in the launcher script,
+                        // it will terminate without connecting back.
+                        while (!_connected.WaitOne(1000) && !proc.ExitCode.HasValue) {
```

Comment:
I think we can get a WaitHandle for the process, which means we can just wait on both of them with `WaitHandle.WaitAny`.

Reply:
Yep, changing it.